<robot name="robot">

	<link name="base_link">
		<visual>
			<geometry>
				<box size="0.16 0.16 0.12"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.085"/>
			<material name="white">
				<color rgba="1 1 1 1"/>
			</material>
		</visual>
		<collision>
			<!-- origin is relative -->
			<origin xyz="0 0 0"/>
			<geometry>
				<box size="0.16 0.16 0.12"/>
			</geometry>
		</collision>
	</link>
	
	<link name="lwheel">
		<visual>
			<geometry>
				<cylinder length="0.008" radius="0.046"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<material name="black">
				<color rgba="0 0 0 1"/>
			</material>
		</visual>
		<collision>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<geometry>
				<cylinder radius="0.046" length="0.008"/>
			</geometry>
		</collision>
	</link>
	
	<joint name="base_to_lwheel" type="fixed">
		<parent link="base_link"/>
		<child link="lwheel"/>
		<origin xyz="0 0.115 0.046" rpy="1.5708 0 0"/>
	</joint>
	
  <transmission name="lwheel_trans" type="SimpleTransmission">
    <actuator name="lwheel_motor"/>
    <joint name="base_to_lwheel"/>
    <mechanicalReduction>1</mechanicalReduction>
    <motorTorqueConstant>1</motorTorqueConstant>
  </transmission>
  
	<link name="rwheel">
		<visual>
			<geometry>
				<cylinder length="0.008" radius="0.046"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<material name="black">
				<color rgba="0 0 0 1"/>
			</material>
		</visual>
		<collision>
			<origin xyz="0 0 0" rpy="0 0 0"/>
			<geometry>
				<cylinder radius="0.046" length="0.008"/>
			</geometry>
		</collision>
	</link>
	
	<joint name="base_to_rwheel" type="fixed">
		<parent link="base_link"/>
		<child link="rwheel"/>
		<origin xyz="0 -0.115 0.046" rpy="-1.5708 0 0"/>
	</joint>
	
  <transmission name="rwheel_trans" type="SimpleTransmission">
    <actuator name="rwheel_motor" />
    <joint name="base_to_rwheel" />
    <mechanicalReduction>1</mechanicalReduction>
    <motorTorqueConstant>1</motorTorqueConstant>
  </transmission>	
	
	<link name="fwheel_attach">
		<visual>
			<geometry>
				<cylinder length="0.060" radius="0.023"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.13 0 0.08"/>
			<material name="white">
				<color rgba="1 1 1 1"/>
			</material>
		</visual>
	</link>
	
	<joint name="base_to_fattach" type="fixed">
		<parent link="base_link"/>
		<child link="fwheel_attach"/>
		<origin xyz="0 0 0" rpy="0 0 0"/>
	</joint>
	
	<link name="fwheel">
		<visual>
			<geometry>
				<cylinder length="0.005" radius="0.020"/>
			</geometry>
			<origin rpy="1.578 0 0" xyz="0 0 0"/>
			<material name="black">
				<color rgba="0 0 0 1"/>
			</material>
		</visual>
	</link>
	
	<joint name="base_to_fwheel" type="fixed">
		<parent link="base_link"/>
		<child link="fwheel"/>
		<origin xyz="0.13 0 0.02" rpy="0 0 0"/>
	</joint>
</robot>